src/dlgConnectionProfiles.cpp

```diff
@@ -1696,20 +1768,79 @@ void dlgConnectionProfiles::slot_copy_profile()
+    newProfiledir.mkpath(newProfiledir.absolutePath());
+    QStringList entries = oldProfiledir.entryList(QDir::Files | QDir::NoDotAndDotDot, QDir::Time);
+    if (entries.empty()) {
         return;
     }
 
-    copyFolder(mudlet::getMudletPath(mudlet::profileHomePath, oldname),
-               mudlet::getMudletPath(mudlet::profileHomePath, profile_name));
-    mProfileList << profile_name;
-    slot_item_clicked(pItem);
-    // Clear the Discord optin on the copied profile - just because the source
-    // one may have had it enabled does not mean we can assume the new one would
-    // want it set:
-    discord_optin_checkBox->setChecked(false);
+    auto copySettingsFromFile = oldProfiledir.absoluteFilePath(entries.first());
+
+    pugi::xml_document newProfileXml;
+    if (extractSettingsFromProfile(newProfileXml, copySettingsFromFile)) {
+        saveProfileCopy(newProfiledir, newProfileXml);
+    }
+
+    return;
+}
+
+bool dlgConnectionProfiles::extractSettingsFromProfile(pugi::xml_document& newProfile, const QString& copySettingsFrom)
+{
+    pugi::xml_document oldProfile;
+    pugi::xml_parse_result result = oldProfile.load_file(copySettingsFrom.toUtf8().constData());
+    if (!result) {
+        qWarning() << "dlgConnectionProfiles::copyProfileSettingsOnly() ERROR: couldn't parse" << copySettingsFrom;
+        qWarning() << "Parse error: " << result.description() << ", character pos= " << result.offset;
+        return false;
+    }
+
+    // write header
+    auto declaration = newProfile.prepend_child(pugi::node_declaration);
+    declaration.append_attribute("version") = "1.0";
+    declaration.append_attribute("encoding") = "UTF-8";
+    newProfile.append_child(pugi::node_doctype).set_value("MudletPackage");
+
+    // copy /MudletPackage attributes
+    auto mudletPackage = newProfile.append_child("MudletPackage");
+    const auto attributeNodes = oldProfile.select_nodes("/MudletPackage/attribute::*");
+    for (pugi::xpath_node_set::const_iterator it = attributeNodes.begin(); it != attributeNodes.end(); ++it) {
+        auto node = *it;
+        mudletPackage.append_attribute(node.attribute().name()) = node.attribute().value();
+    }
+
+    // remove installed packages/modules
+    const auto hostPackageResults = oldProfile.select_nodes("/MudletPackage/HostPackage");
+    pugi::xml_node hostPackage = hostPackageResults.first().node();
+    auto host = hostPackage.child("Host");
+    host.remove_child("mInstalledPackages");
+    host.remove_child("mInstalledModules");
+
+    // copy in the /Mudlet/HostPackage
+    mudletPackage.append_copy(hostPackage);
+    return true;
+}
+
+// save profile using Qt's API's which handle non-ASCII characters in Windows paths fine
+void dlgConnectionProfiles::saveProfileCopy(const QDir& newProfiledir, const pugi::xml_document& newProfileXml) const
+{
+    QFile file(newProfiledir.absoluteFilePath(QStringLiteral("Copied profile (settings only).xml")));
```

Comment:
So the profile settings are copied to that file... Is Mudlet automatically loading all XML files if it can't find the profile.xml? Not sure how this works...

Reply:
Yes, it'll load this file OK and it'll have this in profile history:

![Workspace 1_195](https://user-images.githubusercontent.com/110988/61295764-15620d00-a7d9-11e9-83f0-9c6b4aada661.png)

We need the custom file name here so we can know that if there's only 1 xml file and it's that name, we should install default scripts into it :)